workflow/scripts/s05_cell_metrics_processing.py

```diff
@@ -0,0 +1,220 @@
+
+(sinput, soutput) = getSnake(locals(), 'workflow/spikesort.smk',
+ [cell_metrics_processing_done_path], 'cell_metrics_processing')
+
+
+# %% Define path of processed probes
+
+verbose = True
+sorter_name = 'kilosort3'
+ks3_path = Path(sinput.ephys_sync_complete).parent / sorter_name
+rec_properties_path = Path(sinput.rec_properties)
+
+session_ID = ks3_path.parent.parent.stem
+
+processed_folder = rec_properties_path.parent.parent / 'processed' 
+# Only select probe folders where the results of the sorting can be found.
+probe_folders = [processed_folder / sorter_name / probe_folder / 'sorter_output'
+                  for probe_folder in os.listdir(processed_folder / sorter_name)
+                  if 'spike_clusters.npy' in os.listdir(processed_folder / sorter_name / probe_folder / 'sorter_output')]
+probe_names = [probe_folder.parent.stem for probe_folder in probe_folders]
+
+rec_properties = pd.read_csv(rec_properties_path, header=0, index_col=0)
+# Filter only longest syncable files
+rec_properties = rec_properties[(rec_properties['syncable'] == True) & (rec_properties['longest'] == True)]
+experiments_nb = rec_properties.exp_nb.unique()
+
+# Folder where to store outputs of waveforms info
+waveform_results_folder = rec_properties_path.parent.parent / 'processed' / 'waveforms'
+if not waveform_results_folder.exists():
+    waveform_results_folder.mkdir()
+    
+# %% Get the path of CellExplorer outputs
+
+session_si_df = pd.DataFrame()
+
+for probe_folder in probe_folders:
+    probe_name = probe_folder.parent.stem
+
+    mat_files = list(probe_folder.glob('*.mat'))
+
+    cell_metrics_path = [mat_file for mat_file in mat_files if ('cell_metrics.cellinfo' in str(mat_file))][0]
+    spikes_path = [mat_file for mat_file in mat_files if ('spikes.cellinfo' in str(mat_file))][0]
+
+    # Load .mat files of CellExplorer outputs
+    cell_metrics_dict = loadmat(cell_metrics_path)
+    cell_metrics = cell_metrics_dict['cell_metrics']
+    spikes_dict = loadmat(spikes_path)
+    spikes = spikes_dict['spikes']
+
+    # Get name and shapes of variables
+
+    spikes_var_names = np.array(spikes.dtype.names)
+    cell_var_names = np.array(cell_metrics.dtype.names)
+    nb_clusters = max(cell_metrics['UID'][0][0][0])
+
+    spikes_var_shapes = [spikes[spikes_var_name][0][0][0].shape for spikes_var_name in spikes_var_names]
+    cell_metrics_shapes = [cell_metrics[cell_var_name][0][0][0].shape for cell_var_name in cell_var_names]
+
+    # A lot of reformatting is needed to get from matlab structure to tabular DataFrame
+    # Here we only take varables which have a single value by cluster
+
+    uni_dim_vars_idx = [idx for idx, shape in enumerate(cell_metrics_shapes) if shape == (nb_clusters,)]
+    cell_metrics_df = pd.DataFrame(index = cell_metrics['UID'][0][0][0], columns = cell_var_names[uni_dim_vars_idx])
+    
+    
+    for col in cell_var_names[uni_dim_vars_idx]:
```

Comment:
please don't implement complex analysis in the workflow script directly. It undermines code reuse.